### PR TITLE
refactor: Remove unused nchaintx from SnapshotMetadata constructor, fix test, add test

### PR DESCRIPTION
**Pull request by maflcko**
See commit messages

### CODE REVIEW

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -2694,9 +2694,7 @@ UniValue CreateUTXOSnapshot(
     result.pushKV("base_height", tip->nHeight);
     result.pushKV("path", path.u8string());
     result.pushKV("txoutset_hash", maybe_stats->hashSerialized.ToString());
-    // Cast required because univalue doesn't have serialization specified for
-    // `unsigned int`, nChainTx's type.
-    result.pushKV("nchaintx", uint64_t{tip->nChainTx});
+    result.pushKV("nchaintx", tip->nChainTx);
```
Comment (Sjors) [inline src/rpc/blockchain.cpp:2697]: faa90f6e7b6b8c531e1ae142a5c2f568b48502a9: was this always wrong or is there some recent change to UniValue?

File: test/functional/feature_assumeutxo.py [hunk 2]
```diff
@@ -68,23 +68,35 @@ def setup_network(self):
...
         bad_snapshot_height = SNAPSHOT_BASE_HEIGHT - 1
-        bad_snapshot_path = valid_snapshot_path + '.mod'
         with open(bad_snapshot_path, 'wb') as f:
             bad_snapshot_block_hash = self.nodes[0].getblockhash(bad_snapshot_height)
             # block hash of the snapshot base is stored right at the start (first 32 bytes)
             f.write(bytes.fromhex(bad_snapshot_block_hash)[::-1] + valid_snapshot_contents[32:])
 
         expected_log = f"assumeutxo height in snapshot metadata not recognized ({bad_snapshot_height}) - refusing to load snapshot"
-        with self.nodes[1].assert_debug_log(expected_log):
+        with self.nodes[1].assert_debug_log([expected_log]):
```
Comment (Sjors) [inline test/functional/feature_assumeutxo.py:85]: fafde92f84fb7c245bc3c1cd946a32c891861e5e: I verified this test indeed would pass regardless of `expected_log`. Wen type safety?

**Comment by Sjors** [comment]
utACK fafde92f84fb7c245bc3c1cd946a32c891861e5e

File: test/functional/feature_assumeutxo.py [hunk 3]
```diff
@@ -68,23 +68,35 @@ def setup_network(self):
...
             assert_raises_rpc_error(-32603, "Unable to load UTXO snapshot", self.nodes[1].loadtxoutset, bad_snapshot_path)
 
+        self.log.info("  - snapshot file with wrong number of coins")
+        valid_num_coins = struct.unpack("<I", valid_snapshot_contents[32:32 + 4])[0]
```
Comment (theStack) [inline test/functional/feature_assumeutxo.py:89]: nit: could do the same without the `struct` module by using `int.from_bytes` (and `int.to_bytes` below):
```suggestion
        valid_num_coins = int.from_bytes(valid_snapshot_contents[32:32 + 4], 'little')
```
(mostly a matter of taste I guess though)

**Review by theStack — APPROVED** [review]
ACK fafde92f84fb7c245bc3c1cd946a32c891861e5e

Thanks for fixing the sneaky `assert_debug_log` bug.

**Pull request merged**